diff --git a/extensions/browser/extension_prefs.cc b/extensions/browser/extension_prefs.cc
index 8fa89d4bd502a..1a40bf5d285b6 100644
--- a/extensions/browser/extension_prefs.cc
+++ b/extensions/browser/extension_prefs.cc
@@ -2297,7 +2297,7 @@ void ExtensionPrefs::RegisterProfilePrefs(
   registry->RegisterBooleanPref(pref_names::kAppFullscreenAllowed, true);
 #endif
 
-  registry->RegisterBooleanPref(pref_names::kBlockExternalExtensions, false);
+  registry->RegisterBooleanPref(pref_names::kBlockExternalExtensions, true);
   registry->RegisterIntegerPref(pref_names::kExtensionUnpublishedAvailability,
                                 0);
   registry->RegisterListPref(pref_names::kExtensionInstallTypeBlocklist);
