diff --git a/content/public/common/content_features.cc b/content/public/common/content_features.cc
index a0ac946fc64b4..bb3ba83bf77a9 100644
--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -192,7 +192,7 @@ BASE_FEATURE(kCdmStorageDatabaseMigration,
 // swap BrowsingContextGroups(BrowsingInstances).
 BASE_FEATURE(kClearCrossSiteCrossBrowsingContextGroupWindowName,
              "ClearCrossSiteCrossBrowsingContextGroupWindowName",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kCompositeBGColorAnimation,
              "CompositeBGColorAnimation",
diff --git a/third_party/blink/renderer/core/loader/document_loader.cc b/third_party/blink/renderer/core/loader/document_loader.cc
index 85c4d912e2fdb..d03099ab283f0 100644
--- a/third_party/blink/renderer/core/loader/document_loader.cc
+++ b/third_party/blink/renderer/core/loader/document_loader.cc
@@ -2854,7 +2854,7 @@ void DocumentLoader::CommitNavigation() {
     // that the name would be nulled and if the name is accessed after we will
     // fire a UseCounter. If we decide to move forward with this change, we'd
     // actually clean the name here.
-    // frame_->tree().setName(g_null_atom);
+    frame_->Tree().SetName(g_null_atom);
     frame_->Tree().ExperimentalSetNulledName();
   }
 
@@ -2865,6 +2865,7 @@ void DocumentLoader::CommitNavigation() {
     // TODO(shuuran): CrossSiteCrossBrowsingContextGroupSetNulledName will just
     // record the fact that the name would be nulled and if the name is accessed
     // after we will fire a UseCounter.
+    frame_->Tree().SetName(g_null_atom);
     frame_->Tree().CrossSiteCrossBrowsingContextGroupSetNulledName();
   }
 
