diff --git a/chrome/browser/sharing_hub/sharing_hub_features.cc b/chrome/browser/sharing_hub/sharing_hub_features.cc
index 2ddc318a2b738..6fd447d8e5ce1 100644
--- a/chrome/browser/sharing_hub/sharing_hub_features.cc
+++ b/chrome/browser/sharing_hub/sharing_hub_features.cc
@@ -72,7 +72,7 @@ BASE_FEATURE(kDesktopScreenshots,
 
 #if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_CHROMEOS)
 void RegisterProfilePrefs(PrefRegistrySimple* registry) {
-  registry->RegisterBooleanPref(prefs::kDesktopSharingHubEnabled, true);
+  registry->RegisterBooleanPref(prefs::kDesktopSharingHubEnabled, false);
 }
 #endif
 
diff --git a/chrome/browser/ui/ui_features.cc b/chrome/browser/ui/ui_features.cc
index 703fc03b7e6cc..60347901f0d21 100644
--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -35,7 +35,7 @@ BASE_FEATURE(kWebAppIconInTitlebar,
              base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Enables Chrome Labs menu in the toolbar. See https://crbug.com/1145666
-BASE_FEATURE(kChromeLabs, "ChromeLabs", base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kChromeLabs, "ChromeLabs", base::FEATURE_DISABLED_BY_DEFAULT);
 const char kChromeLabsActivationParameterName[] =
     "chrome_labs_activation_percentage";
 const base::FeatureParam<int> kChromeLabsActivationPercentage{
@@ -280,11 +280,7 @@ BASE_FEATURE(kTabGroupsCollapseFreezing,
 // https://crbug.com/928954
 BASE_FEATURE(kTabHoverCardImages,
              "TabHoverCardImages",
-#if BUILDFLAG(IS_MAC)
              base::FEATURE_DISABLED_BY_DEFAULT
-#else
-             base::FEATURE_ENABLED_BY_DEFAULT
-#endif
 );
 
 const char kTabHoverCardImagesNotReadyDelayParameterName[] =
diff --git a/components/history_clusters/core/features.cc b/components/history_clusters/core/features.cc
index a2244cdbd0e7e..9058d86b5e8d7 100644
--- a/components/history_clusters/core/features.cc
+++ b/components/history_clusters/core/features.cc
@@ -26,7 +26,7 @@ constexpr auto enabled_by_default_desktop_only =
 
 namespace internal {
 
-BASE_FEATURE(kJourneys, "Journeys", enabled_by_default_desktop_only);
+BASE_FEATURE(kJourneys, "Journeys", base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kJourneysImages,
              "JourneysImages",
