diff --git a/content/browser/site_info.h b/content/browser/site_info.h
index 743c414a5d4ad..02a312a7522b1 100644
--- a/content/browser/site_info.h
+++ b/content/browser/site_info.h
@@ -482,7 +482,7 @@ class CONTENT_EXPORT SiteInfo {
   bool does_site_request_dedicated_process_for_coop_ = false;
 
   // Indicates that JIT is disabled for this SiteInfo.
-  bool is_jit_disabled_ = false;
+  bool is_jit_disabled_ = true;
 
   // Indicates that v8 optimizations are disabled for this SiteInfo.
   bool are_v8_optimizations_disabled_ = false;
