diff --git a/content/browser/site_info.cc b/content/browser/site_info.cc
index 0a6be043ddffe..a9a0f7ef62e1e 100644
--- a/content/browser/site_info.cc
+++ b/content/browser/site_info.cc
@@ -229,7 +229,7 @@ SiteInfo SiteInfo::CreateInternal(const IsolationContext& isolation_context,
 
   // PDF content should live in JIT-less processes because it is inherently less
   // trusted.
-  bool is_jitless = url_info.is_pdf;
+  bool is_jitless = true;
   bool are_v8_optimizations_disabled = false;
 
   std::optional<StoragePartitionConfig> storage_partition_config =
