diff --git a/chrome/browser/browser_features.cc b/chrome/browser/browser_features.cc
index d9f547ea6ef9b..65241fceaba89 100644
--- a/chrome/browser/browser_features.cc
+++ b/chrome/browser/browser_features.cc
@@ -15,6 +15,10 @@
 
 namespace features {
 
+// Disable V8's JIT on local pages (doesn't affect webpages)
+BASE_FEATURE(kDisableV8JIT, "DisableV8JIT",
+             base::FEATURE_ENABLED_BY_DEFAULT);
+
 #if BUILDFLAG(IS_ANDROID)
 // Kill switch for allowing TWAs to autoplay with sound without requiring a user
 // gesture to unlock, for parity with PWAs.
diff --git a/chrome/browser/browser_features.h b/chrome/browser/browser_features.h
index 18c5d790e8413..acd6e8626cb04 100644
--- a/chrome/browser/browser_features.h
+++ b/chrome/browser/browser_features.h
@@ -16,6 +16,8 @@
 
 namespace features {
 
+BASE_DECLARE_FEATURE(kDisableV8JIT);
+
 // All features in alphabetical order. The features should be documented
 // alongside the definition of their values in the .cc file.
 
diff --git a/chrome/browser/chrome_content_browser_client.cc b/chrome/browser/chrome_content_browser_client.cc
index ac83f5229cea0..c03e287ceee20 100644
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -2740,6 +2740,10 @@ void MaybeAppendBlinkSettingsSwitchForFieldTrial(
 void ChromeContentBrowserClient::AppendExtraCommandLineSwitches(
     base::CommandLine* command_line,
     int child_process_id) {
+  if (base::FeatureList::IsEnabled(features::kDisableV8JIT)) {
+    command_line->AppendSwitchASCII(blink::switches::kJavaScriptFlags,
+                                                     "--jitless");
+  }
   crash_keys::AppendStringAnnotationsCommandLineSwitch(command_line);
 #if BUILDFLAG(IS_MAC)
   std::unique_ptr<metrics::ClientInfo> client_info =
diff --git a/content/browser/renderer_host/render_process_host_impl.cc b/content/browser/renderer_host/render_process_host_impl.cc
index ba9f41f010a37..97e0c8fb83ecb 100644
--- a/content/browser/renderer_host/render_process_host_impl.cc
+++ b/content/browser/renderer_host/render_process_host_impl.cc
@@ -3192,6 +3192,10 @@ void RenderProcessHostImpl::AppendRendererCommandLine(
   } else if (AreV8OptimizationsDisabled()) {
     command_line->AppendSwitchASCII(blink::switches::kJavaScriptFlags,
                                     "--disable-optimizing-compilers");
+  } else {
+    // Allow for per-site exceptions even when JIT is globally disabled,
+    // the "dummy value" here allows the override
+    command_line->AppendSwitchASCII(blink::switches::kJavaScriptFlags, "");
   }
 
   if (features::IsTouchTextEditingRedesignEnabled()) {
