diff --git a/chrome/browser/net/system_network_context_manager.cc b/chrome/browser/net/system_network_context_manager.cc
index 249ff5ecffa8d..c9c36e3226290 100644
--- a/chrome/browser/net/system_network_context_manager.cc
+++ b/chrome/browser/net/system_network_context_manager.cc
@@ -533,8 +533,12 @@ void SystemNetworkContextManager::GssapiLibraryLoadObserver::Install(
 
 void SystemNetworkContextManager::GssapiLibraryLoadObserver::
     OnBeforeGssapiLibraryLoad() {
+  // Keeping this enabled will disable the Network Service Sandbox when a
+  // website tries to use GSSAPI, not very secure. Flag can re-enable.
   owner_->local_state_->SetBoolean(prefs::kReceivedHttpAuthNegotiateHeader,
-                                   true);
+                                   base::CommandLine::
+                                   ForCurrentProcess()->HasSwitch(
+                                                        "enable-gssapi"));
 }
 #endif  // BUILDFLAG(IS_LINUX)
 
diff --git a/services/network/public/mojom/network_service.mojom b/services/network/public/mojom/network_service.mojom
index 3f3dac717be42..29a9946052308 100644
--- a/services/network/public/mojom/network_service.mojom
+++ b/services/network/public/mojom/network_service.mojom
@@ -95,7 +95,11 @@ struct HttpAuthDynamicParams {
 
   // Indicates whether the GSSAPI library should be loaded. Only supported on
   // Chrome OS and Linux.
-  bool allow_gssapi_library_load = true;
+  // GSSAPI will disable the Network Service Sandbox when websites try to load
+  // it, not desirable from a security perspective.
+  bool allow_gssapi_library_load = base::CommandLine::
+                                   ForCurrentProcess()->HasSwitch(
+                                                        "enable-gssapi"));
 
   // True if Basic authentication challenges should be allowed for non-secure
   // HTTP responses.
