diff --git a/remoting/host/policy_watcher.cc b/remoting/host/policy_watcher.cc
index f7274c6a02f29..660fc9690e9b1 100644
--- a/remoting/host/policy_watcher.cc
+++ b/remoting/host/policy_watcher.cc
@@ -180,7 +180,7 @@ base::Value::Dict PolicyWatcher::GetPlatformPolicies() {
 
 base::Value::Dict PolicyWatcher::GetDefaultPolicies() {
   base::Value::Dict result;
-  result.Set(key::kRemoteAccessHostFirewallTraversal, true);
+  result.Set(key::kRemoteAccessHostFirewallTraversal, false);
   result.Set(key::kRemoteAccessHostClientDomainList, base::Value::List());
   result.Set(key::kRemoteAccessHostDomainList, base::Value::List());
   result.Set(key::kRemoteAccessHostAllowRelayedConnection, true);
   result.Set(key::kRemoteAccessHostUdpPortRange, "");
   result.Set(key::kRemoteAccessHostClipboardSizeBytes, -1);
-  result.Set(key::kRemoteAccessHostAllowRemoteSupportConnections, true);
+  result.Set(key::kRemoteAccessHostAllowRemoteSupportConnections, false);
 #if BUILDFLAG(IS_CHROMEOS)
   result.Set(key::kRemoteAccessHostAllowEnterpriseRemoteSupportConnections,
              true);
@@ -206,7 +206,7 @@ base::Value::Dict PolicyWatcher::GetDefaultPolicies() {
   result.Set(key::kRemoteAccessHostAllowFileTransfer, true);
   result.Set(key::kRemoteAccessHostAllowUrlForwarding, true);
   result.Set(key::kRemoteAccessHostEnableUserInterface, true);
-  result.Set(key::kRemoteAccessHostAllowRemoteAccessConnections, true);
+  result.Set(key::kRemoteAccessHostAllowRemoteAccessConnections, false);
   result.Set(key::kRemoteAccessHostMaximumSessionDurationMinutes, 0);
   result.Set(key::kRemoteAccessHostAllowPinAuthentication, base::Value());
 #endif
