diff --git a/content/browser/renderer_host/render_process_host_impl.cc b/content/browser/renderer_host/render_process_host_impl.cc
index ba9f41f010a37..28571feba231e 100644
--- a/content/browser/renderer_host/render_process_host_impl.cc
+++ b/content/browser/renderer_host/render_process_host_impl.cc
@@ -3192,6 +3192,12 @@ void RenderProcessHostImpl::AppendRendererCommandLine(
   } else if (AreV8OptimizationsDisabled()) {
     command_line->AppendSwitchASCII(blink::switches::kJavaScriptFlags,
                                     "--disable-optimizing-compilers");
+  } else {
+    // Allow for per-site exceptions even when JIT is globally disabled,
+    // the "dummy value" here just enables WASM which should already be
+    // enabled so no harm done
+    command_line->AppendSwitchASCII(blink::switches::kJavaScriptFlags,
+                                    "--expose-wasm");
   }
 
   if (features::IsTouchTextEditingRedesignEnabled()) {
diff --git a/gin/gin_features.cc b/gin/gin_features.cc
index e7f829d0d5ddb..6fb45e0cf6eeb 100644
--- a/gin/gin_features.cc
+++ b/gin/gin_features.cc
@@ -8,6 +8,10 @@
 
 namespace features {
 
+// Disable V8's JIT on all pages (doesn't affect webpages)
+BASE_FEATURE(kDisableV8JIT, "DisableV8JIT",
+             base::FEATURE_ENABLED_BY_DEFAULT);
+
 // Enable code space compaction when finalizing a full GC with stack.
 BASE_FEATURE(kV8CompactCodeSpaceWithStack,
              ("V8CompactCodeSpaceWithStack"),
diff --git a/gin/gin_features.h b/gin/gin_features.h
index ae743c8c4194f..9143c07ba973e 100644
--- a/gin/gin_features.h
+++ b/gin/gin_features.h
@@ -20,6 +20,8 @@ namespace features {
 inline constexpr base::FeatureState kFeatureDefaultStateControlledByV8 =
     base::FEATURE_DISABLED_BY_DEFAULT;
 
+GIN_EXPORT BASE_DECLARE_FEATIRE(kDisableV8JIT);
+
 GIN_EXPORT BASE_DECLARE_FEATURE(kV8CompactCodeSpaceWithStack);
 GIN_EXPORT BASE_DECLARE_FEATURE(kV8CompactWithStack);
 GIN_EXPORT BASE_DECLARE_FEATURE(kV8ConcurrentSparkplug);
diff --git a/gin/v8_initializer.cc b/gin/v8_initializer.cc
index d08f79157897c..bef6a0d9d2f84 100644
--- a/gin/v8_initializer.cc
+++ b/gin/v8_initializer.cc
@@ -222,6 +222,10 @@ void SetV8FlagsIfOverridden(const base::Feature& feature,
 
 void SetFlags(IsolateHolder::ScriptMode mode,
               const std::string js_command_line_flags) {
+  if (base::FeatureList::IsEnabled(features::kDisableV8JIT)) {
+    SetV8Flags("--jitless");
+  }
+
   // We assume that all feature flag defaults correspond to the default
   // values of the corresponding V8 flags.
   SetV8FlagsIfOverridden(features::kV8CompactCodeSpaceWithStack,
