diff --git a/components/variations/service/variations_service.cc b/components/variations/service/variations_service.cc
index 914ecd9f396c2..58dda94779e10 100644
--- a/components/variations/service/variations_service.cc
+++ b/components/variations/service/variations_service.cc
@@ -226,17 +226,7 @@ bool GetInstanceManipulations(const net::HttpResponseHeaders* headers,
 // Variations seed fetching is only enabled in official Chrome builds, if a URL
 // is specified on the command line, and for testing.
 bool IsFetchingEnabled() {
-#if !BUILDFLAG(GOOGLE_CHROME_BRANDING)
-  if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
-          switches::kVariationsServerURL) &&
-      !g_should_fetch_for_testing) {
-    DVLOG(1)
-        << "Not performing repeated fetching in unofficial build without --"
-        << switches::kVariationsServerURL << " specified.";
-    return false;
-  }
-#endif
-  return true;
+  return false;
 }
 
 // Returns the already downloaded first run seed, and clear the seed from the
@@ -551,10 +541,10 @@ void VariationsService::RegisterPrefs(PrefRegistrySimple* registry) {
 
   registry->RegisterIntegerPref(
       prefs::kDeviceVariationsRestrictionsByPolicy,
-      static_cast<int>(RestrictionPolicy::NO_RESTRICTIONS));
+      static_cast<int>(RestrictionPolicy::ALL));
   registry->RegisterDictionaryPref(
       prefs::kVariationsGoogleGroups,
-      static_cast<int>(RestrictionPolicy::NO_RESTRICTIONS));
+      static_cast<int>(RestrictionPolicy::ALL));
   // This preference keeps track of the country code used to filter
   // permanent-consistency studies.
   registry->RegisterListPref(prefs::kVariationsPermanentConsistencyCountry);
@@ -566,7 +556,7 @@ void VariationsService::RegisterPrefs(PrefRegistrySimple* registry) {
   // allows the admin to restrict the set of variations applied.
   registry->RegisterIntegerPref(
       prefs::kVariationsRestrictionsByPolicy,
-      static_cast<int>(RestrictionPolicy::NO_RESTRICTIONS));
+      static_cast<int>(RestrictionPolicy::ALL));
   // This preference will only be written by the policy service, which will fill
   // it according to a value stored in the User Policy.
   registry->RegisterStringPref(prefs::kVariationsRestrictParameter,
