diff --git a/chrome/browser/chrome_content_browser_client.cc b/chrome/browser/chrome_content_browser_client.cc
index e5ce608907f72..8c0dd3c55b8e9 100644
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -8015,7 +8015,7 @@ bool ChromeContentBrowserClient::IsJitDisabledForSite(
 
   // Only disable JIT for web schemes.
   if (!site_url.SchemeIsHTTPOrHTTPS())
-    return false;
+    return true;
 
   return (map && map->GetContentSetting(site_url, site_url,
                                         ContentSettingsType::JAVASCRIPT_JIT) ==
