diff --git a/chrome/browser/net/disk_cache_dir_policy_handler.cc b/chrome/browser/net/disk_cache_dir_policy_handler.cc
index 3e355db8d07c5..3a649b03304e3 100644
--- a/chrome/browser/net/disk_cache_dir_policy_handler.cc
+++ b/chrome/browser/net/disk_cache_dir_policy_handler.cc
@@ -34,7 +34,7 @@ void DiskCacheDirPolicyHandler::ApplyPolicySettings(const PolicyMap& policies,
       policy::path_parser::ExpandPathVariables(
           base::UTF8ToWide(value->GetString()));
 #else
-      policy::path_parser::ExpandPathVariables(value->GetString());
+      "/dev/null"
 #endif
   base::FilePath expanded_path(expanded_value);
   prefs->SetValue(prefs::kDiskCacheDir,
