diff --git a/chrome/browser/devtools/devtools_window.cc b/chrome/browser/devtools/devtools_window.cc
index a7c33cf116281..25bafa2243b90 100644
--- a/chrome/browser/devtools/devtools_window.cc
+++ b/chrome/browser/devtools/devtools_window.cc
@@ -529,7 +529,7 @@ void DevToolsWindow::RegisterProfilePrefs(
       prefs::kDevToolsSyncedPreferencesSyncDisabled);
   registry->RegisterIntegerPref(
       prefs::kDevToolsGenAiSettings,
-      static_cast<int>(DevToolsGenAiEnterprisePolicyValue::kAllow));
+      static_cast<int>(DevToolsGenAiEnterprisePolicyValue::kDisable));
 }
 
 // static
diff --git a/components/optimization_guide/core/model_execution/model_execution_prefs.cc b/components/optimization_guide/core/model_execution/model_execution_prefs.cc
index d33df93b9ad75..b61239d2cd221 100644
--- a/components/optimization_guide/core/model_execution/model_execution_prefs.cc
+++ b/components/optimization_guide/core/model_execution/model_execution_prefs.cc
@@ -30,19 +30,19 @@ const char kHistorySearchEnterprisePolicyAllowed[] =
 void RegisterProfilePrefs(PrefRegistrySimple* registry) {
   registry->RegisterIntegerPref(
       kTabOrganizationEnterprisePolicyAllowed,
-      static_cast<int>(ModelExecutionEnterprisePolicyValue::kAllow),
+      static_cast<int>(ModelExecutionEnterprisePolicyValue::kDisable),
       PrefRegistry::LOSSY_PREF);
   registry->RegisterIntegerPref(
       kComposeEnterprisePolicyAllowed,
-      static_cast<int>(ModelExecutionEnterprisePolicyValue::kAllow),
+      static_cast<int>(ModelExecutionEnterprisePolicyValue::kDisable),
       PrefRegistry::LOSSY_PREF);
   registry->RegisterIntegerPref(
       kWallpaperSearchEnterprisePolicyAllowed,
-      static_cast<int>(ModelExecutionEnterprisePolicyValue::kAllow),
+      static_cast<int>(ModelExecutionEnterprisePolicyValue::kDisable),
       PrefRegistry::LOSSY_PREF);
   registry->RegisterIntegerPref(
       kHistorySearchEnterprisePolicyAllowed,
-      static_cast<int>(ModelExecutionEnterprisePolicyValue::kAllow),
+      static_cast<int>(ModelExecutionEnterprisePolicyValue::kDisable),
       PrefRegistry::LOSSY_PREF);
 }
 
@@ -138,7 +138,7 @@ void RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
   registry->RegisterInt64Pref(localstate::kModelQualityLogggingClientId, 0,
                               PrefRegistry::LOSSY_PREF);
   registry->RegisterIntegerPref(
-      localstate::kGenAILocalFoundationalModelEnterprisePolicySettings, 0);
+      localstate::kGenAILocalFoundationalModelEnterprisePolicySettings, 1);
 }
 
 // LINT.IfChange(GetOnDeviceFeatureRecentlyUsedPref)
