diff --git a/content/public/common/content_features.cc b/content/public/common/content_features.cc
index 75b2720edef81..2763ad5a33c7d 100644
--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -47,7 +47,7 @@ BASE_FEATURE(kAudioServiceOutOfProcess,
 // kAudioServiceOutOfProcess feature is enabled.
 BASE_FEATURE(kAudioServiceSandbox,
              "AudioServiceSandbox",
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_FUCHSIA)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_FUCHSIA) || BUILDFLAG(IS_LINUX)
              base::FEATURE_ENABLED_BY_DEFAULT
 #else
              base::FEATURE_DISABLED_BY_DEFAULT
