diff --git a/third_party/blink/common/features.cc b/third_party/blink/common/features.cc
index 16e843fd62d0a..400602645bf2f 100644
--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -1749,7 +1749,7 @@ const base::FeatureParam<bool> kPartialLowEndModeExcludeCanvasFontCache{
 // TODO(crbug.com/329102369): complete the partitioned hashtable implementation.
 BASE_FEATURE(kPartitionVisitedLinkDatabase,
              "PartitionVisitedLinkDatabase",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Enables the use of the PaintCache for Path2D objects that are rasterized
 // out of process.  Has no effect when kCanvasOopRasterization is disabled.
