diff --git a/third_party/blink/common/features.cc b/third_party/blink/common/features.cc
index 00e268b511265..ccfca911e3a2f 100644
--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -2081,7 +2081,7 @@ BASE_FEATURE(kSaveDataImgSrcset,
 // fetched it was the same as the current document.
 BASE_FEATURE(kScopeMemoryCachePerContext,
              "ScopeMemoryCachePerContext",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kPausePagesPerBrowsingContextGroup,
              "PausePagesPerBrowsingContextGroup",
