diff --git a/extensions/common/extension_features.cc b/extensions/common/extension_features.cc
index b598c4723199a..2597912500296 100644
--- a/extensions/common/extension_features.cc
+++ b/extensions/common/extension_features.cc
@@ -100,7 +100,7 @@ BASE_FEATURE(kExtensionWebFileHandlers,
 
 BASE_FEATURE(kExtensionsManifestV3Only,
              "ExtensionsManifestV3Only",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kExtensionsMenuAccessControl,
              "ExtensionsMenuAccessControl",
