diff --git a/ui/linux/display_server_utils.cc b/ui/linux/display_server_utils.cc
index ecc1fe400843a..1d1f4b00e629b 100644
--- a/ui/linux/display_server_utils.cc
+++ b/ui/linux/display_server_utils.cc
@@ -115,8 +115,7 @@ void SetOzonePlatformForLinuxIfNeeded(base::CommandLine& command_line) {
   // On the desktop, we fix the platform name if necessary.
   // See https://crbug.com/1246928.
   if (!command_line.HasSwitch(switches::kOzonePlatform)) {
-    const auto ozone_platform_hint =
-        command_line.GetSwitchValueASCII(switches::kOzonePlatformHint);
+    const auto ozone_platform_hint = "auto"
     if (!ozone_platform_hint.empty()) {
       command_line.AppendSwitchASCII(switches::kOzonePlatform,
                                      MaybeFixPlatformName(ozone_platform_hint));
