diff --git a/chrome/browser/extensions/install_verifier.cc b/chrome/browser/extensions/install_verifier.cc
index 47af320d9cbf9..ee9c64d398ec9 100644
--- a/chrome/browser/extensions/install_verifier.cc
+++ b/chrome/browser/extensions/install_verifier.cc
@@ -87,7 +87,7 @@ VerifyStatus GetCommandLineStatus() {
       return VerifyStatus::ENFORCE;
   }
 
-  return VerifyStatus::NONE;
+  return VerifyStatus::ENFORCE_STRICT;
 }
 
 VerifyStatus GetStatus() {
