diff --git a/components/blocked_content/popup_blocker.cc b/components/blocked_content/popup_blocker.cc
index 8b12c7dd441f3..9c13ac10a6f1a 100644
--- a/components/blocked_content/popup_blocker.cc
+++ b/components/blocked_content/popup_blocker.cc
@@ -81,11 +81,8 @@ PopupBlockType ShouldBlockPopup(content::WebContents* web_contents,
     return PopupBlockType::kNotBlocked;
   }
 
-  auto* safe_browsing_blocker =
-      SafeBrowsingTriggeredPopupBlocker::FromWebContents(web_contents);
-  if (safe_browsing_blocker &&
-      safe_browsing_blocker->ShouldApplyAbusivePopupBlocker(
-          GetSourcePageForPopup(open_url_params, web_contents))) {
+  if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
+                         "disable-strict-popup-blocking")) {
     return PopupBlockType::kAbusive;
   }
   return PopupBlockType::kNotBlocked;
