diff --git a/components/cbor/BUILD.gn b/components/cbor/BUILD.gn
index 91fbfffd3c..f8f8a4c01f 100644
--- a/components/cbor/BUILD.gn
+++ b/components/cbor/BUILD.gn
@@ -12,7 +12,7 @@ buildflag_header("buildflags") {
   # TODO(crbug.com/535682335): Remove `USE_CBOR_RUST` buildflag entirely,
   # unconditionally enable Rust CBOR parser, and drop `is_cronet_build` check
   # once Cronet supports Crubit dependencies.
-  flags = [ "USE_CBOR_RUST=!$is_cronet_build" ]
+  flags = [ "USE_CBOR_RUST=false" ]
 }

 component("cbor") {
@@ -36,13 +36,6 @@ component("cbor") {
     ":buildflags",
     "//base",
   ]
-
-  if (!is_cronet_build) {
-    public_deps = [
-      "//build/rust/crubit",
-      "//components/cbor/rust:cbor_rust_bindings",
-    ]
-  }
 }

 source_set("unit_tests") {
