diff --git a/chromium.conf b/chromium.conf
index 62cecce..b51f9d7 100644
--- a/chromium.conf
+++ b/chromium.conf
@@ -9,7 +9,14 @@ GRAPHIC_DRIVER=default
 WEB_DARKMODE=off
 
 # NATIVE_WAYLAND=[on|off]
-NATIVE_WAYLAND=off
+case "$XDG_SESSION_TYPE" in
+   wayland)
+      NATIVE_WAYLAND=on
+      ;;
+   x11)
+      NATIVE_WAYLAND=off
+      ;;
+esac
 
 CHROMIUM_FLAGS=""
 if [ "$NATIVE_WAYLAND" == "off" ] ; then
@@ -17,7 +24,6 @@ if [ "$NATIVE_WAYLAND" == "off" ] ; then
    CHROMIUM_FLAGS+=" --enable-gpu-memory-buffer-video-frames"
    CHROMIUM_FLAGS+=" --enable-zero-copy"
    CHROMIUM_FLAGS+=" --ignore-gpu-blocklist --disable-gpu-driver-bug-workaround"
-   CHROMIUM_FLAGS+=" --enable-chrome-browser-cloud-management"
    CHROMIUM_FLAGS+=" --enable-gpu-rasterization"
 fi
 
--- a/chromium.spec
+++ b/chromium.spec
@@ -36,10 +36,10 @@
 	ninja -j %{numjobs} -C '%1' '%2'
 
 # enable|disable chromedriver
-%global build_chromedriver 1
+%global build_chromedriver 0
 
 # enable|disable headless client build
-%global build_headless 1
+%global build_headless 0
 %ifarch ppc64le
 %global build_headless 0
 %endif
@@ -113,7 +113,7 @@
 %global userestrictedapikeys 0
 
 # We can still use the api key though. For now.
-%global useapikey 1
+%global useapikey 0
 
 # Leave this alone, please.
 %global chromebuilddir out/Release
@@ -295,8 +295,8 @@
 %global chromoting_client_id %nil
 %endif
 
-Name:	chromium%{chromium_channel}
-Version: 127.0.6533.99
+Name:	hardened-chromium%{chromium_channel}
+Version: 128.0.6613.84
 Release: 1%{?dist}
 Summary: A WebKit (Blink) powered web browser that Google doesn't want you to use
 Url: http://www.chromium.org/Home
@@ -479,7 +479,92 @@ Patch503: chromium-127-ninja-1.21.1-deps-part2.patch
 Patch504: chromium-127-ninja-1.21.1-deps-part3.patch
 Patch505: chromium-127-crabbyavif.patch
 Patch506: chromium-127-allow-enabling-vulkan-on-ozone-wayland.patch
 
+# hardening patches
+
+Patch2000: disable-various-content-settings-by-default.patch
+Patch2001: block-external-extensions.patch
+Patch2002: disable-autofill-by-default.patch
+Patch2003: disable-background-mode-by-default.patch
+Patch2004: set-browser-defaults.patch
+Patch2005: disable-disk-cache.patch
+Patch2006: disable-metrics-reporting.patch
+Patch2007: disable-promotions-by-default.patch
+Patch2008: disable-remote-access-by-default.patch
+Patch2009: disable-search-suggest-by-default.patch
+Patch2010: enable-audio-service-sandbox.patch
+Patch2011: restrict-default-supported-http-auth-schemes.patch
+Patch2012: disable-variations.patch
+Patch2013: set-default-always-open-pdf-externally.patch
+Patch2014: set-default-extension-content-verification-enforce-strict.patch
+Patch2015: set-default-extension-install-verification-enforce-strict.patch
+Patch2016: set-default-secure-dns-mode-automatic.patch
+Patch2017: set-mv3-only-by-default.patch
+Patch2018: set-ozone-platform-hint-auto-by-default.patch
+Patch2019: show-full-urls-by-default.patch
+Patch2020: disable-password-manager-prompt-by-default.patch
+Patch2021: enable-network-service-sandbox-by-default.patch
+Patch2022: disable-sync-by-default.patch
+Patch2023: disable-infobar-for-builds-without-api-key.patch
+Patch2024: disable-printing-by-default.patch
+Patch2025: enable-visited-link-database-partitioning.patch
+Patch2026: disable-tpcd-bypasses.patch
+Patch2027: disable-gen-ai-features-and-logging-by-default.patch
+Patch2028: disable-secondary-browser-features-by-default.patch
+Patch2029: remove-unused-preferences.patch
+Patch2030: add-cross-origin-referrer-clearing-feature.patch
+Patch2031: expose-flags.patch
+Patch2032: hide-profile-icon-feature.patch
+Patch2033: add-feature-to-show-puny-code.patch
+
+# Vanadium patches
+
+Patch3000: 0011-Checkout-PGO-profiles.patch
+Patch3001: 0012-disable-checkout_nacl.patch
+Patch3002: 0014-disable-seed-based-field-trials.patch
+Patch3004: 0017-disable-navigation-error-correction-by-default.patch
+Patch3006: 0019-disable-network-prediction-by-default.patch
+Patch3007: 0021-disable-hyperlink-auditing-by-default.patch
+Patch3008: 0022-disable-showing-popular-sites-by-default.patch
+Patch3009: 0023-disable-article-suggestions-feature-by-default.patch
+Patch3010: 0024-disable-content-feed-suggestions-by-default.patch
+Patch3011: 0025-disable-sensors-access-by-default.patch
+Patch3012: 0026-block-playing-protected-media-by-default.patch
+Patch3013: 0027-disable-third-party-cookies-by-default.patch
+Patch3014: 0028-disable-background-sync-by-default.patch
+Patch3015: 0029-disable-payment-support-by-default.patch
+Patch3016: 0030-disable-media-router-media-remoting-by-default.patch
+Patch3017: 0031-disable-media-router-by-default.patch
+Patch3019: 0033-disable-browser-sign-in-feature-by-default.patch
+Patch3021: 0035-disable-safe-browsing-reporting-opt-in-by-default.patch
+Patch3022: 0036-disable-unused-safe-browsing-option-by-default.patch
+Patch3023: 0037-disable-media-DRM-preprovisioning-by-default.patch
+Patch3024: 0038-disable-autofill-server-communication-by-default.patch
+Patch3025: 0039-disable-component-updater-pings-by-default.patch
+Patch3027: 0041-disable-trivial-subdomain-hiding.patch
+Patch3029: 0044-disable-GaiaAuthFetcher-code-due-to-upstream-bug.patch
+Patch3030: 0046-Disable-newer-privacy-sandbox-features-by-default.patch
+Patch3031: 0048-Disable-top-toolbar-button-Translate-option-by-defau.patch
+Patch3032: 0049-always-use-local-new-tab-page.patch
+Patch3033: 0050-mark-non-secure-origins-as-dangerous.patch
+Patch3035: 0052-stub-out-the-battery-status-API.patch
+Patch3036: 0055-disable-trials-of-privacy-aware-analytics-advertisin.patch
+Patch3037: 0057-disable-appending-variations-header.patch
+Patch3038: 0058-Disable-detailed-language-settings-by-default.patch
+Patch3039: 0059-disable-fetching-optimization-guides-by-default.patch
+Patch3041: 0063-require-HTTPS-for-component-updates.patch
+Patch3042: 0069-enable-prefetch-privacy-changes-by-default.patch
+Patch3043: 0070-enable-split-cache-by-default.patch
+Patch3044: 0071-enable-partitioning-connections-by-default.patch
+Patch3045: 0074-Enable-strict-origin-isolation-by-default.patch
+Patch3046: 0075-Enable-reduce-accept-language-header-by-default.patch
+Patch3048: 0077-use-Google-Chrome-branding-for-client-hints.patch
+Patch3049: 0115-Derive-high-entropy-client-hints-with-reduced-user-a.patch
+Patch3050: 0122-Use-local-list-of-supported-languages-for-Language-s.patch
+Patch3051: 0165-Enable-content-settings-partitioning-by-default.patch
+Patch3052: 0170-Revert-M122-content-make-disable-jit-actually-disabl.patch
+Patch3053: 0084-temporary-Always-partition-third-party-storage.patch
+Patch3054: 0183-Isolate-sandboxed-iframes-per-site-by-default.patch
 # Use chromium-latest.py to generate clean tarball from released build tarballs, found here:
 # http://build.chromium.org/buildbot/official/
 # For Chromium Fedora use chromium-latest.py --stable --ffmpegclean --ffmpegarm
@@ -501,21 +586,6 @@ Source9: chromium-browser.xml
 Source10: chrome-remote-desktop@.service
 Source11: master_preferences
 
-%if ! %{system_nodejs}
-Source12: https://nodejs.org/dist/%{nodejs_version}/node-%{nodejs_version}-linux-x64.tar.xz
-Source13: https://nodejs.org/dist/%{nodejs_version}/node-%{nodejs_version}-linux-arm64.tar.xz
-%endif
-
-# esbuild binary
-%if 0%{?rhel}
-Source14: https://registry.npmjs.org/@esbuild/linux-x64/-/linux-x64-%{esbuild_version}.tgz
-Source15: https://registry.npmjs.org/@esbuild/linux-arm64/-/linux-arm64-%{esbuild_version}.tgz
-%endif
-
-# bindgen for epel8
-Source16: https://than.fedorapeople.org/epel8/bindgen-cli-aarch64.tar.xz
-Source17: https://than.fedorapeople.org/epel8/bindgen-cli-x86_64.tar.xz
-
 # esbuild binary from fedora
 %if 0%{?fedora}
 BuildRequires: golang-github-evanw-esbuild
@@ -847,7 +917,7 @@ Requires: libcanberra-gtk3%{_isa}
 Requires: u2f-hidraw-policy
 %endif
 
-Requires: chromium-common%{_isa} = %{version}-%{release}
+Requires: hardened-chromium-common%{_isa} = %{version}-%{release}
 
 # rhel 8 or newer and fedora < 40: x86_64, aarch64
 # fedora 40 or newer: x86_64, aarch64, ppc64le
@@ -1012,7 +1082,7 @@ Requires(preun): systemd
 Requires(postun): systemd
 Requires: xorg-x11-server-Xvfb
 Requires: python3-psutil
-Requires: chromium-common%{_isa} = %{version}-%{release}
+Requires: hardened-chromium-common%{_isa} = %{version}-%{release}
 Summary: Remote desktop support for google-chrome & chromium
 
 %description -n chrome-remote-desktop
@@ -1021,7 +1091,7 @@ Remote desktop support for google-chrome & chromium.
 
 %package -n chromedriver
 Summary: WebDriver for Google Chrome/Chromium
-Requires: chromium-common%{_isa} = %{version}-%{release}
+Requires: hardened-chromium-common%{_isa} = %{version}-%{release}
 
 %description -n chromedriver
 WebDriver is an open source tool for automated testing of webapps across many
@@ -1032,7 +1102,7 @@ members of the Chromium and WebDriver teams.
 
 %package headless
 Summary:	A minimal headless shell built from Chromium
-Requires: chromium-common%{_isa} = %{version}-%{release}
+Requires: hardened-chromium-common%{_isa} = %{version}-%{release}
 
 %description headless
 A minimal headless client built from Chromium. headless_shell is built
@@ -1041,14 +1111,14 @@ udev.
 
 %package qt5-ui
 Summary: Qt5 UI built from Chromium
-Requires: chromium%{_isa} = %{version}-%{release}
+Requires: hardened-chromium%{_isa} = %{version}-%{release}
 
 %description qt5-ui
 Qt5 UI for chromium.
 
 %package qt6-ui
 Summary: Qt6 UI built from Chromium
-Requires: chromium%{_isa} = %{version}-%{release}
+Requires: hardened-chromium%{_isa} = %{version}-%{release}
 
 %description qt6-ui
 Qt6 UI for chromium.
@@ -1213,7 +1283,9 @@ Qt6 UI for chromium.
 %endif
 %patch -P505 -p1 -b .crabbyavif
 %patch -P506 -p1 -b .allow-enabling-vulkan-on-ozone-wayland
 
+%autopatch -m 2000 -M 2033
+%autopatch -m 3000 -M 3054
 # Change shebang in all relevant files in this directory and all subdirectories
 # See `man find` for how the `-exec command {} +` syntax works
 find -type f \( -iname "*.py" \) -exec sed -i '1s=^#! */usr/bin/\(python\|env python\)[23]\?=#!%{chromium_pybin}=' {} +
@@ -1279,7 +1351,7 @@ cp -a %{_includedir}/libusb-1.0/libusb.h third_party/libusb/src/libusb/libusb.h
 %endif
 
 # Hard code extra version
-sed -i 's/getenv("CHROME_VERSION_EXTRA")/"Fedora Project"/' chrome/common/channel_info_posix.cc
+sed -i 's/getenv("CHROME_VERSION_EXTRA")/"hardened-chromium"/' chrome/common/channel_info_posix.cc
 
 # Fix hardcoded path in remoting code
 sed -i 's|/opt/google/chrome-remote-desktop|%{crd_path}|g' remoting/host/setup/daemon_controller_delegate_linux.cc
@@ -1375,11 +1447,11 @@ sed -i 's|OFFICIAL_BUILD|GOOGLE_CHROME_BUILD|g' tools/generate_shim_headers/gene
 
 CHROMIUM_CORE_GN_DEFINES+=' chrome_pgo_phase=0'
 
-%if ! %{cfi}
-CHROMIUM_CORE_GN_DEFINES+=' is_cfi=false use_thin_lto=false'
-%else
 CHROMIUM_CORE_GN_DEFINES+=' is_cfi=true use_thin_lto=true'
-%endif
+
+CHROMIUM_CORE_GN_DEFINES+=' enable_reporting=false'
+
+CHROMIUM_CORE_GN_DEFINES+=' enable_remoting=false'
 
 %if %{useapikey}
 CHROMIUM_CORE_GN_DEFINES+=' google_api_key="%{api_key}"'
@@ -1425,6 +1497,9 @@ CHROMIUM_CORE_GN_DEFINES+=' use_custom_libcxx=false'
 %endif
 CHROMIUM_CORE_GN_DEFINES+=' enable_iterator_debugging=false'
 CHROMIUM_CORE_GN_DEFINES+=' enable_vr=false'
+CHROMIUM_CORE_GN_DEFINES+=' enable_arcore=false'
+CHROMIUM_CORE_GN_DEFINES+=' enable_openxr=false'
+CHROMIUM_CORE_GN_DEFINES+=' enable_cardboard=false'
 CHROMIUM_CORE_GN_DEFINES+=' build_dawn_tests=false enable_perfetto_unittests=false'
 CHROMIUM_CORE_GN_DEFINES+=' disable_fieldtrial_testing_config=true'
 CHROMIUM_CORE_GN_DEFINES+=' symbol_level=%{debug_level} blink_symbol_level=%{debug_level}'
@@ -1464,8 +1539,8 @@ CHROMIUM_BROWSER_GN_DEFINES+=' use_qt6=false'
 %endif
 
 CHROMIUM_BROWSER_GN_DEFINES+=' use_gio=true use_pulseaudio=true'
-CHROMIUM_BROWSER_GN_DEFINES+=' enable_hangout_services_extension=true'
-CHROMIUM_BROWSER_GN_DEFINES+=' enable_widevine=true'
+CHROMIUM_BROWSER_GN_DEFINES+=' enable_hangout_services_extension=false'
+CHROMIUM_BROWSER_GN_DEFINES+=' enable_widevine=false'
 
 %if %{use_vaapi}
 CHROMIUM_BROWSER_GN_DEFINES+=' use_vaapi=true'
@@ -1667,15 +1742,15 @@ rm -rf %{buildroot}
 
 mkdir -p %{buildroot}%{_bindir} \
          %{buildroot}%{chromium_path}/locales \
-         %{buildroot}%{_sysconfdir}/%{name}
+         %{buildroot}%{_sysconfdir}/chromium
 
 # install system wide chromium config
-cp -a %{SOURCE2} %{buildroot}%{_sysconfdir}/%{name}/%{name}.conf
+cp -a %{SOURCE2} %{buildroot}%{_sysconfdir}/chromium/chromium.conf
 cp -a %{SOURCE3} %{buildroot}%{chromium_path}/%{chromium_browser_channel}.sh
 
 %if ! %{use_vaapi}
 # remove vaapi flags
-echo "# system wide chromium flags" > %{buildroot}%{_sysconfdir}/%{name}/%{name}.conf
+echo "# system wide chromium flags" > %{buildroot}%{_sysconfdir}/chromium/chromium.conf
 %endif
 
 export BUILD_TARGET=`cat /etc/redhat-release`
@@ -1836,7 +1911,7 @@ mkdir -p %{buildroot}%{_datadir}/icons/hicolor/24x24/apps
 cp -a chrome/app/theme/chromium/product_logo_24.png %{buildroot}%{_datadir}/icons/hicolor/24x24/apps/%{chromium_browser_channel}.png
 
 # Install the master_preferences file
-install -m 0644 %{SOURCE11} %{buildroot}%{_sysconfdir}/%{name}/
+install -m 0644 %{SOURCE11} %{buildroot}%{_sysconfdir}/chromium/
 
 mkdir -p %{buildroot}%{_datadir}/applications/
 desktop-file-install --dir %{buildroot}%{_datadir}/applications %{SOURCE4}
@@ -1888,11 +1963,11 @@ getent group chrome-remote-desktop >/dev/null || groupadd -r chrome-remote-deskt
 %doc AUTHORS README.fedora
 %doc chrome_policy_list.html *.json
 %license LICENSE
-%config(noreplace) %{_sysconfdir}/%{name}/chromium.conf
-%config %{_sysconfdir}/%{name}/master_preferences
-%config %{_sysconfdir}/%{name}/policies/
+%config(noreplace) %{_sysconfdir}/chromium/chromium.conf
+%config %{_sysconfdir}/chromium/master_preferences
+%config %{_sysconfdir}/chromium/policies/
 %if %{build_remoting}
-%exclude %{_sysconfdir}/%{name}/native-messaging-hosts/*
+%exclude %{_sysconfdir}/chromium/native-messaging-hosts/*
 %endif
 %{_bindir}/%{chromium_browser_channel}
 %{chromium_path}/*.bin
--- a/master_preferences
+++ b/master_preferences
@@ -1,5 +1,5 @@
 {
-  "homepage": "https://fedoraproject.org/start",
+  "homepage": "about:blank",
   "homepage_is_newtabpage": false,
   "distribution": {
      "alternate_shortcut_text": false,
@@ -11,6 +11,6 @@
      "verbose_logging": false
   },
   "first_run_tabs": [
-     "https://fedoraproject.org/start"
+     "about:blank"
   ]
 }
--- a/sources
+++ /dev/null
@@ -1,7 +0,0 @@
-SHA512 (node-v20.6.1-linux-arm64.tar.xz) = adfcaf2c22614797fd69fb46d94c1cbf64dea0213cc817c45d3904b634dbf1f4e62e4ebd95bfa4ba0a9c559747d42115406edc471af294334160ba6e103e31d0
-SHA512 (node-v20.6.1-linux-x64.tar.xz) = 7e15c05041a9a50f0046266aadb2e092a5aefbec19be1c7c809471add520cb57c7df3c47d88b1888b29bf2979dca3c92adddfd965370fa2a9da4ea02186464fd
-SHA512 (linux-arm64-0.19.2.tgz) = 8a0d8fec6786fffcd6954d00820037a55d61e60762c74300df0801f8db27057562c221a063bedfb8df56af9ba80abb366336987e881782c5996e6f871abd3dc6
-SHA512 (linux-x64-0.19.2.tgz) = a31cc74c4bfa54f9b75d735a1cfc944d3b5efb7c06bfba9542da9a642ae0b2d235ea00ae84d3ad0572c406405110fe7b61377af0fd15803806ef78d20fc6f05d
-SHA512 (bindgen-cli-aarch64.tar.xz) = 1a5ae4e8fdd31d80e8111c4d5f2115336684763ecd3a442ffecdbc2a37bab146f88bdee0bb1ea7a98e1049f81b12e64bd0ce5510529b30a74ce3306488ac129b
-SHA512 (bindgen-cli-x86_64.tar.xz) = 7ccc9b43b32d3a064a75cfc150e060711356da8fe98e83d855bae017108ef8e9e172fbdd6e2579433c19cfb56ababa5b77a8db6fa57a5e657a3878778ca10a37
-SHA512 (chromium-127.0.6533.99-clean.tar.xz) = 353658ea97df8f1428e6b8b7b4eb2a18d9f2b21196af0e5bfe37928859ba494a43c2e1a93fdc3fd35a884d92cf75864b6acfa0d7c90c62c6d5fe25c1daeffa10
