diff --git a/chromium.spec b/chromium.spec
index 931b9c0..451591e 100644
--- a/chromium.spec
+++ b/chromium.spec
@@ -31,10 +31,10 @@
 	ninja -j %{numjobs} -C '%1' '%2'
 
 # enable|disable chromedriver
-%global build_chromedriver 1
+%global build_chromedriver 0
 
 # enable|disable headless client build
-%global build_headless 1
+%global build_headless 0
 %ifarch ppc64le
 %global build_headless 0
 %endif
@@ -123,7 +123,7 @@
 %global userestrictedapikeys 0
 
 # We can still use the api key though. For now.
-%global useapikey 1
+%global useapikey 0
 
 # Leave this alone, please.
 %global chromebuilddir out/Release
@@ -309,9 +309,9 @@
 %global chromoting_client_id %nil
 %endif
 
-Name:	chromium%{chromium_channel}
+Name:	hardened-chromium%{chromium_channel}
 Version: 126.0.6478.182
-Release: 1%{?dist}
+Release: 3%{?dist}
 Summary: A WebKit (Blink) powered web browser that Google doesn't want you to use
 Url: http://www.chromium.org/Home
 License: BSD-3-Clause AND LGPL-2.1-or-later AND Apache-2.0 AND IJG AND MIT AND GPL-2.0-or-later AND ISC AND OpenSSL AND (MPL-1.1 OR GPL-2.0-only OR LGPL-2.0-only)
@@ -563,6 +563,83 @@ Patch413: fix-unknown-warning-option-messages.diff
 # 64kpage support on el8
 Patch500: chromium-124-el8-support-64kpage.patch
 
+# hardening patches
+
+Patch2000: 0011-Checkout-PGO-profiles.patch
+Patch2001: 0012-disable-checkout_nacl.patch
+Patch2002: 0014-disable-seed-based-field-trials.patch
+Patch2004: 0017-disable-navigation-error-correction-by-default.patch
+Patch2006: 0019-disable-network-prediction-by-default.patch
+Patch2007: 0021-disable-hyperlink-auditing-by-default.patch
+Patch2008: 0022-disable-showing-popular-sites-by-default.patch
+Patch2009: 0023-disable-article-suggestions-feature-by-default.patch
+Patch2010: 0024-disable-content-feed-suggestions-by-default.patch
+Patch2011: 0025-disable-sensors-access-by-default.patch
+Patch2012: 0026-block-playing-protected-media-by-default.patch
+Patch2013: 0027-disable-third-party-cookies-by-default.patch
+Patch2014: 0028-disable-background-sync-by-default.patch
+Patch2015: 0029-disable-payment-support-by-default.patch
+Patch2016: 0030-disable-media-router-media-remoting-by-default.patch
+Patch2017: 0031-disable-media-router-by-default.patch
+Patch2019: 0033-disable-browser-sign-in-feature-by-default.patch
+Patch2020: 0034-disable-browser-autologin-by-default.patch
+Patch2021: 0035-disable-safe-browsing-reporting-opt-in-by-default.patch
+Patch2022: 0036-disable-unused-safe-browsing-option-by-default.patch
+Patch2023: 0037-disable-media-DRM-preprovisioning-by-default.patch
+Patch2024: 0038-disable-autofill-server-communication-by-default.patch
+Patch2025: 0039-disable-component-updater-pings-by-default.patch
+Patch2027: 0041-disable-trivial-subdomain-hiding.patch
+Patch2029: 0044-disable-GaiaAuthFetcher-code-due-to-upstream-bug.patch
+Patch2030: 0046-Disable-newer-privacy-sandbox-features-by-default.patch
+Patch2031: 0048-Disable-top-toolbar-button-Translate-option-by-defau.patch
+Patch2032: 0049-always-use-local-new-tab-page.patch
+Patch2033: 0050-mark-non-secure-origins-as-dangerous.patch
+Patch2035: 0052-stub-out-the-battery-status-API.patch
+Patch2036: 0055-disable-trials-of-privacy-aware-analytics-advertisin.patch
+Patch2037: 0057-disable-appending-variations-header.patch
+Patch2038: 0058-Disable-detailed-language-settings-by-default.patch
+Patch2039: 0059-disable-fetching-optimization-guides-by-default.patch
+Patch2041: 0063-require-HTTPS-for-component-updates.patch
+Patch2042: 0069-enable-prefetch-privacy-changes-by-default.patch
+Patch2043: 0070-enable-split-cache-by-default.patch
+Patch2044: 0071-enable-partitioning-connections-by-default.patch
+Patch2045: 0074-Enable-strict-origin-isolation-by-default.patch
+Patch2046: 0075-Enable-reduce-accept-language-header-by-default.patch
+Patch2047: 0076-enable-process-isolated-sandboxed-iframes-by-default.patch
+Patch2048: 0077-use-Google-Chrome-branding-for-client-hints.patch
+Patch2049: 0115-Derive-high-entropy-client-hints-with-reduced-user-a.patch
+Patch2050: 0122-Use-local-list-of-supported-languages-for-Language-s.patch
+Patch2051: 0165-Enable-content-settings-partitioning-by-default.patch
+Patch2052: 0170-Revert-M122-content-make-disable-jit-actually-disabl.patch
+Patch2053: 0084-temporary-Always-partition-third-party-storage.patch
+Patch2054: disable-javascript-jit-by-default.patch
+Patch2055: block-external-extensions.patch
+Patch2056: disable-autofill-by-default.patch
+Patch2057: disable-background-mode-by-default.patch
+Patch2058: set-browser-defaults.patch
+Patch2059: disable-disk-cache.patch
+Patch2060: disable-metrics-reporting.patch
+Patch2061: disable-promotions-by-default.patch
+Patch2062: disable-remote-access-by-default.patch
+Patch2063: disable-search-suggest-by-default.patch
+Patch2064: enable-audio-service-sandbox.patch
+Patch2065: restrict-default-supported-http-auth-schemes.patch
+Patch2066: set-critical-variations-only.patch
+Patch2067: set-default-always-open-pdf-externally.patch
+Patch2068: set-default-extension-content-verification-enforce-strict.patch
+Patch2069: set-default-extension-install-verification-enforce-strict.patch
+Patch2070: set-default-secure-dns-mode-automatic.patch
+Patch2071: set-mv3-only-by-default.patch
+Patch2072: set-ozone-platform-hint-auto-by-default.patch
+Patch2073: show-full-urls-by-default.patch
+Patch2074: disable-password-manager-prompt-by-default.patch
+Patch2075: enable-network-service-sandbox-by-default.patch
+Patch2076: disable-sync-by-default.patch
+Patch2077: disable-infobar-for-builds-without-api-key.patch
+Patch2078: disable-printing-by-default.patch
+Patch2079: enable-visited-link-database-partitioning.patch
+Patch2080: disable-tpcd-bypasses.patch
+
 # Use chromium-latest.py to generate clean tarball from released build tarballs, found here:
 # http://build.chromium.org/buildbot/official/
 # For Chromium Fedora use chromium-latest.py --stable --ffmpegclean --ffmpegarm
@@ -928,7 +1005,7 @@ Requires: libcanberra-gtk3%{_isa}
 Requires: u2f-hidraw-policy
 %endif
 
-Requires: chromium-common%{_isa} = %{version}-%{release}
+Requires: hardened-chromium-common%{_isa} = %{version}-%{release}
 
 # rhel 7: x86_64
 # rhel 8 or newer: x86_64, aarch64
@@ -1098,7 +1175,7 @@ Requires(preun): systemd
 Requires(postun): systemd
 Requires: xorg-x11-server-Xvfb
 Requires: python3-psutil
-Requires: chromium-common%{_isa} = %{version}-%{release}
+Requires: hardened-chromium-common%{_isa} = %{version}-%{release}
 Summary: Remote desktop support for google-chrome & chromium
 
 %description -n chrome-remote-desktop
@@ -1107,7 +1184,7 @@ Remote desktop support for google-chrome & chromium.
 
 %package -n chromedriver
 Summary: WebDriver for Google Chrome/Chromium
-Requires: chromium-common%{_isa} = %{version}-%{release}
+Requires: hardened-chromium-common%{_isa} = %{version}-%{release}
 
 %description -n chromedriver
 WebDriver is an open source tool for automated testing of webapps across many
@@ -1118,7 +1195,7 @@ members of the Chromium and WebDriver teams.
 
 %package headless
 Summary:	A minimal headless shell built from Chromium
-Requires: chromium-common%{_isa} = %{version}-%{release}
+Requires: hardened-chromium-common%{_isa} = %{version}-%{release}
 
 %description headless
 A minimal headless client built from Chromium. headless_shell is built
@@ -1127,14 +1204,14 @@ udev.
 
 %package qt5-ui
 Summary: Qt5 UI built from Chromium
-Requires: chromium%{_isa} = %{version}-%{release}
+Requires: hardened-chromium%{_isa} = %{version}-%{release}
 
 %description qt5-ui
 Qt5 UI for chromium.
 
 %package qt6-ui
 Summary: Qt6 UI built from Chromium
-Requires: chromium%{_isa} = %{version}-%{release}
+Requires: hardened-chromium%{_isa} = %{version}-%{release}
 
 %description qt6-ui
 Qt6 UI for chromium.
@@ -1341,6 +1418,80 @@ cp /opt/rh/%{toolset}-%{dts_version}/root/usr/include/c++/%{dts_version}/optiona
 %endif
 %endif
 
+%patch -P2000 -p1 -b .0011-Checkout-PGO-profiles
+%patch -P2001 -p1 -b .0012-disable-checkout_nacl
+%patch -P2002 -p1 -b .0014-disable-seed-based-field-trials
+%patch -P2004 -p1 -b .0017-disable-navigation-error-correction-by-default
+%patch -P2006 -p1 -b .0019-disable-network-prediction-by-default
+%patch -P2007 -p1 -b .0021-disable-hyperlink-auditing-by-default
+%patch -P2008 -p1 -b .0022-disable-showing-popular-sites-by-default
+%patch -P2009 -p1 -b .0023-disable-article-suggestions-feature-by-default
+%patch -P2010 -p1 -b .0024-disable-content-feed-suggestions-by-default
+%patch -P2011 -p1 -b .0025-disable-sensors-access-by-default
+%patch -P2012 -p1 -b .0026-block-playing-protected-media-by-default
+%patch -P2013 -p1 -b .0027-disable-third-party-cookies-by-default
+%patch -P2014 -p1 -b .0028-disable-background-sync-by-default
+%patch -P2015 -p1 -b .0029-disable-payment-support-by-default
+%patch -P2016 -p1 -b .0030-disable-media-router-media-remoting-by-default
+%patch -P2017 -p1 -b .0031-disable-media-router-by-default
+%patch -P2019 -p1 -b .0033-disable-browser-sign-in-feature-by-default
+%patch -P2020 -p1 -b .0034-disable-browser-autologin-by-default
+%patch -P2021 -p1 -b .0035-disable-safe-browsing-reporting-opt-in-by-default
+%patch -P2022 -p1 -b .0036-disable-unused-safe-browsing-option-by-default
+%patch -P2023 -p1 -b .0037-disable-media-DRM-preprovisioning-by-default
+%patch -P2024 -p1 -b .0038-disable-autofill-server-communication-by-default
+%patch -P2025 -p1 -b .0039-disable-component-updater-pings-by-default
+%patch -P2027 -p1 -b .0041-disable-trivial-subdomain-hiding
+%patch -P2029 -p1 -b .0044-disable-GaiaAuthFetcher-code-due-to-upstream-bug
+%patch -P2030 -p1 -b .0046-Disable-newer-privacy-sandbox-features-by-default
+%patch -P2031 -p1 -b .0048-Disable-top-toolbar-button-Translate-option-by-defau
+%patch -P2032 -p1 -b .0049-always-use-local-new-tab-page
+%patch -P2033 -p1 -b .0050-mark-non-secure-origins-as-dangerous
+%patch -P2035 -p1 -b .0052-stub-out-the-battery-status-API
+%patch -P2036 -p1 -b .0055-disable-trials-of-privacy-aware-analytics-advertisin
+%patch -P2037 -p1 -b .0057-disable-appending-variations-header
+%patch -P2038 -p1 -b .0058-Disable-detailed-language-settings-by-default
+%patch -P2039 -p1 -b .0059-disable-fetching-optimization-guides-by-default
+%patch -P2041 -p1 -b .0063-require-HTTPS-for-component-updates
+%patch -P2042 -p1 -b .0069-enable-prefetch-privacy-changes-by-default
+%patch -P2043 -p1 -b .0070-enable-split-cache-by-default
+%patch -P2044 -p1 -b .0071-enable-partitioning-connections-by-default
+%patch -P2045 -p1 -b .0074-Enable-strict-origin-isolation-by-default
+%patch -P2046 -p1 -b .0075-Enable-reduce-accept-language-header-by-default
+%patch -P2047 -p1 -b .0076-enable-process-isolated-sandboxed-iframes-by-default
+%patch -P2048 -p1 -b .0077-use-Google-Chrome-branding-for-client-hints
+%patch -P2049 -p1 -b .0115-Derive-high-entropy-client-hints-with-reduced-user-a
+%patch -P2050 -p1 -b .0122-Use-local-list-of-supported-languages-for-Language-s
+%patch -P2051 -p1 -b .0165-Enable-content-settings-partitioning-by-default
+%patch -P2052 -p1 -b .0170-Revert-M122-content-make-disable-jit-actually-disabl
+%patch -P2053 -p1 -b .0084-temporary-Always-partition-third-party-storage
+%patch -P2054 -p1 -b .disable-javascript-jit-by-default
+%patch -P2055 -p1 -b .block-external-extensions
+%patch -P2056 -p1 -b .disable-autofill-by-default
+%patch -P2057 -p1 -b .disable-background-mode-by-default
+%patch -P2058 -p1 -b .set-browser-defaults
+%patch -P2059 -p1 -b .disable-disk-cache
+%patch -P2060 -p1 -b .disable-metrics-reporting
+%patch -P2061 -p1 -b .disable-promotions-by-default
+%patch -P2062 -p1 -b .disable-remote-access-by-default
+%patch -P2063 -p1 -b .disable-search-suggest-by-default
+%patch -P2064 -p1 -b .enable-audio-service-sandbox
+%patch -P2065 -p1 -b .restrict-default-supported-http-auth-schemes
+%patch -P2066 -p1 -b .set-critical-variations-only
+%patch -P2067 -p1 -b .set-default-always-open-pdf-externally
+%patch -P2068 -p1 -b .set-default-extension-content-verification-enforce-strict
+%patch -P2069 -p1 -b .set-default-extension-install-verification-enforce-strict
+%patch -P2070 -p1 -b .set-default-secure-dns-mode-automatic
+%patch -P2071 -p1 -b .set-mv3-only-by-default
+#%patch -P2072 -p1 -b .set-ozone-platform-hint-auto-by-default
+%patch -P2073 -p1 -b .show-full-urls-by-default
+%patch -P2074 -p1 -b .disable-password-manager-prompt-by-default
+%patch -P2075 -p1 -b .enable-network-service-sandbox-by-default
+%patch -P2076 -p1 -b .disable-sync-by-default
+%patch -P2077 -p1 -b .disable-infobar-for-builds-without-api-key
+%patch -P2078 -p1 -b .disable-printing-by-default
+%patch -P2079 -p1 -b .enable-visited-link-database-partitioning
+%patch -P2080 -p1 -b .disable-tpcd-bypasses
 # Change shebang in all relevant files in this directory and all subdirectories
 # See `man find` for how the `-exec command {} +` syntax works
 find -type f \( -iname "*.py" \) -exec sed -i '1s=^#! */usr/bin/\(python\|env python\)[23]\?=#!%{chromium_pybin}=' {} +
@@ -1393,7 +1544,7 @@ cp -a %{_includedir}/libusb-1.0/libusb.h third_party/libusb/src/libusb/libusb.h
 %endif
 
 # Hard code extra version
-sed -i 's/getenv("CHROME_VERSION_EXTRA")/"Fedora Project"/' chrome/common/channel_info_posix.cc
+sed -i 's/getenv("CHROME_VERSION_EXTRA")/"hardened-chromium"/' chrome/common/channel_info_posix.cc
 
 # Fix hardcoded path in remoting code
 sed -i 's|/opt/google/chrome-remote-desktop|%{crd_path}|g' remoting/host/setup/daemon_controller_delegate_linux.cc
@@ -1494,9 +1645,11 @@ sed -i 's|OFFICIAL_BUILD|GOOGLE_CHROME_BUILD|g' tools/generate_shim_headers/gene
 
 CHROMIUM_CORE_GN_DEFINES+=' chrome_pgo_phase=0'
 
-%if ! %{cfi}
-CHROMIUM_CORE_GN_DEFINES+=' is_cfi=false use_thin_lto=false'
-%endif
+CHROMIUM_CORE_GN_DEFINES+=' is_cfi=true use_thin_lto=true'
+
+CHROMIUM_CORE_GN_DEFINES+=' enable_reporting=false'
+
+CHROMIUM_CORE_GN_DEFINES+=' enable_remoting=false'
 
 %if %{useapikey}
 CHROMIUM_CORE_GN_DEFINES+=' google_api_key="%{api_key}"'
@@ -1547,6 +1700,9 @@ CHROMIUM_CORE_GN_DEFINES+=' use_custom_libcxx=false'
 %endif
 CHROMIUM_CORE_GN_DEFINES+=' enable_iterator_debugging=false'
 CHROMIUM_CORE_GN_DEFINES+=' enable_vr=false'
+CHROMIUM_CORE_GN_DEFINES+=' enable_arcore=false'
+CHROMIUM_CORE_GN_DEFINES+=' enable_openxr=false'
+CHROMIUM_CORE_GN_DEFINES+=' enable_cardboard=false'
 CHROMIUM_CORE_GN_DEFINES+=' build_dawn_tests=false enable_perfetto_unittests=false'
 CHROMIUM_CORE_GN_DEFINES+=' disable_fieldtrial_testing_config=true'
 CHROMIUM_CORE_GN_DEFINES+=' symbol_level=%{debug_level} blink_symbol_level=%{debug_level}'
@@ -1586,8 +1742,8 @@ CHROMIUM_BROWSER_GN_DEFINES+=' use_qt6=false'
 %endif
 
 CHROMIUM_BROWSER_GN_DEFINES+=' use_gio=true use_pulseaudio=true'
-CHROMIUM_BROWSER_GN_DEFINES+=' enable_hangout_services_extension=true'
-CHROMIUM_BROWSER_GN_DEFINES+=' enable_widevine=true'
+CHROMIUM_BROWSER_GN_DEFINES+=' enable_hangout_services_extension=false'
+CHROMIUM_BROWSER_GN_DEFINES+=' enable_widevine=false'
 
 %if %{use_vaapi}
 CHROMIUM_BROWSER_GN_DEFINES+=' use_vaapi=true'
@@ -1784,15 +1940,15 @@ rm -rf %{buildroot}
 
 mkdir -p %{buildroot}%{_bindir} \
          %{buildroot}%{chromium_path}/locales \
-         %{buildroot}%{_sysconfdir}/%{name}
+         %{buildroot}%{_sysconfdir}/chromium
 
 # install system wide chromium config
-cp -a %{SOURCE2} %{buildroot}%{_sysconfdir}/%{name}/%{name}.conf
+cp -a %{SOURCE2} %{buildroot}%{_sysconfdir}/chromium/chromium.conf
 cp -a %{SOURCE3} %{buildroot}%{chromium_path}/%{chromium_browser_channel}.sh
 
 %if ! %{use_vaapi}
 # remove vaapi flags
-echo "# system wide chromium flags" > %{buildroot}%{_sysconfdir}/%{name}/%{name}.conf
+echo "# system wide chromium flags" > %{buildroot}%{_sysconfdir}/chromium/chromium.conf
 %endif
 
 export BUILD_TARGET=`cat /etc/redhat-release`
@@ -1953,7 +2109,7 @@ mkdir -p %{buildroot}%{_datadir}/icons/hicolor/24x24/apps
 cp -a chrome/app/theme/chromium/product_logo_24.png %{buildroot}%{_datadir}/icons/hicolor/24x24/apps/%{chromium_browser_channel}.png
 
 # Install the master_preferences file
-install -m 0644 %{SOURCE11} %{buildroot}%{_sysconfdir}/%{name}/
+install -m 0644 %{SOURCE11} %{buildroot}%{_sysconfdir}/chromium/
 
 mkdir -p %{buildroot}%{_datadir}/applications/
 desktop-file-install --dir %{buildroot}%{_datadir}/applications %{SOURCE4}
@@ -2005,11 +2161,11 @@ getent group chrome-remote-desktop >/dev/null || groupadd -r chrome-remote-deskt
 %doc AUTHORS README.fedora
 %doc chrome_policy_list.html *.json
 %license LICENSE
-%config(noreplace) %{_sysconfdir}/%{name}/chromium.conf
-%config %{_sysconfdir}/%{name}/master_preferences
-%config %{_sysconfdir}/%{name}/policies/
+%config(noreplace) %{_sysconfdir}/chromium/chromium.conf
+%config %{_sysconfdir}/chromium/master_preferences
+%config %{_sysconfdir}/chromium/policies/
 %if %{build_remoting}
-%exclude %{_sysconfdir}/%{name}/native-messaging-hosts/*
+%exclude %{_sysconfdir}/chromium/native-messaging-hosts/*
 %endif
 %{_bindir}/%{chromium_browser_channel}
 %{chromium_path}/*.bin
--- a/master_preferences
+++ b/master_preferences
@@ -1,5 +1,5 @@
 {
-  "homepage": "https://fedoraproject.org/start",
+  "homepage": "about:blank",
   "homepage_is_newtabpage": false,
   "distribution": {
      "alternate_shortcut_text": false,
@@ -11,6 +11,6 @@
      "verbose_logging": false
   },
   "first_run_tabs": [
-     "https://fedoraproject.org/start"
+     "about:blank"
   ]
 }
