diff --git a/chromium.spec b/chromium.spec
index 5c6a9ed..cca215e 100644
--- a/chromium.spec
+++ b/chromium.spec
@@ -563,6 +563,9 @@ Patch413: fix-unknown-warning-option-messages.diff
 # 64kpage support on el8
 Patch500: chromium-124-el8-support-64kpage.patch
 
+# hardening patches
+Patch2000: 0025-disable-sensors-access-by-default.patch
+
 # Use chromium-latest.py to generate clean tarball from released build tarballs, found here:
 # http://build.chromium.org/buildbot/official/
 # For Chromium Fedora use chromium-latest.py --stable --ffmpegclean --ffmpegarm
