diff -up chromium-129.0.6668.42.old/media/filters/ffmpeg_glue.cc.me chromium-129.0.6668.42.old/media/filters/ffmpeg_glue.cc
--- chromium-129.0.6668.42.old/media/filters/ffmpeg_glue.cc.me	2024-09-15 22:12:33.945186323 +0200
+++ chromium-129.0.6668.42.old/media/filters/ffmpeg_glue.cc	2024-09-16 10:53:13.334970539 +0200
@@ -17,14 +17,11 @@
 
 namespace media {
 
-// Kill switches in case things explode. Remove after M132.
+// Kill switch in case things explode. Remove after M132.
 // TODO(crbug.com/355485812): Re-enable this flag.
 BASE_FEATURE(kAllowOnlyAudioCodecsDuringDemuxing,
              "AllowOnlyAudioCodecsDuringDemuxing",
              base::FEATURE_ENABLED_BY_DEFAULT);
-BASE_FEATURE(kForbidH264ParsingDuringDemuxing,
-             "ForbidH264ParsingDuringDemuxing",
-             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Internal buffer size used by AVIO for reading.
 // TODO(dalecurtis): Experiment with this buffer size and measure impact on
@@ -118,12 +115,6 @@ FFmpegGlue::FFmpegGlue(FFmpegURLProtocol
   // Enable fast, but inaccurate seeks for MP3.
   format_context_->flags |= AVFMT_FLAG_FAST_SEEK;
 
-  // We don't allow H.264 parsing during demuxing since we have our own parser
-  // and the ffmpeg one increases memory usage unnecessarily.
-  if (base::FeatureList::IsEnabled(kForbidH264ParsingDuringDemuxing)) {
-    format_context_->flags |= AVFMT_FLAG_NOH264PARSE;
-  }
-
   // Ensures format parsing errors will bail out. From an audit on 11/2017, all
   // instances were real failures. Solves bugs like http://crbug.com/710791.
   format_context_->error_recognition |= AV_EF_EXPLODE;
